src/include/transaction/transaction_context.h

```diff
@@ -51,6 +74,26 @@ class TransactionContext {
         undo_buffer_(buffer_pool.Get()),
         redo_buffer_(log_manager.Get(), buffer_pool.Get()) {}
 
+
+  /**
+   * Constructs a new transaction context.
+   *
+   * @warning In the src/ folder this should only be called in TransactionManager::BeginTransaction to adhere to MVCC
+   * semantics. Tests are allowed to deterministically construct them in ways that violate the current MVCC semantics.
+   * @warning Beware that the buffer pool given must be the same one the log manager uses,
+   * if logging is enabled.
+   * @param start the start timestamp of the transaction. Should be unique within the system.
+   * @param finish in HyPer parlance this is txn id. Should be larger than all start times and commit times in current
```

Comment:
what is HyPer parlance? Maybe its bad practice to refer to other DBMSs in comment

Reply:
This is a copy of another TransactionContext constructor comment - I'm wary of removing it in both places since we didn't write it